modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -433,9 +433,30 @@
                     }
                 }
                 else {
-                    if (log.isInfoEnabled())
+                    if (log.isInfoEnabled()) {
+                        String serverNodeId = "";
+                        String consistentId = "";
+
+                        try {
+                            Collection<ClusterNode> remoteNodes = getSpiContext().remoteNodes();
+
+                            for (ClusterNode node : remoteNodes) {
+                                if (U.toInetAddresses(node).contains(ses.remoteAddress().getAddress()) && !node.isClient()) {
+                                    serverNodeId = node.id().toString();
+                                    consistentId = node.consistentId().toString();
+
+                                    break;
+                                }
+                            }
+                        } catch (IgniteCheckedException e) {
+                            log.warning("Unable to acquire remote node ID from SpiContext: " + e, e);
+                        }
+
                         log.info("Established outgoing communication connection [locAddr=" + ses.localAddress() +
```

Comment:
Please wrap `log.info()` to `if(log.isInfoEnabled())` condition.

Reply:
But `log.info()` already wrapped

Reply:
Ok. I look over it.